### ISSUE TITLE
💅 `noUnusedVariables` - false-positive for param destructuring if it's used as a default value for another destructured param

### ISSUE THREAD

**Issue by ipanasenko** [biomejs/biome#1652]
### Environment information

```bash
CLI:
  Version:                      1.5.2
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           macos

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v18.18.2"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "npm/9.8.1"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    true
  VCS disabled:                 true

Workspace:
  Open Documents:               0
```

### Rule name

noUnusedVariables

### Playground link

https://biomejs.dev/playground/?lintRules=all&code=YwBvAG4AcwB0ACAAYQBGAHUAbgBjAHQAaQBvAG4AIAA9ACAAKAB7AGEALAAgAGIAIAA9ACAAYQB9ACkAIAA9AD4AIAB7AAoAIAAgAGMAbwBuAHMAbwBsAGUALgBpAG4AZgBvACgAYgApADsACgB9AAoAYwBvAG4AcwBvAGwAZQAuAGkAbgBmAG8AKABhAEYAdQBuAGMAdABpAG8AbgApADsA

### Expected result

No linting error

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct
